### PR TITLE
Gui: add Game icon and size icon choice in config

### PR DESCRIPTION
**Pull request by Zangetsu38**
thx @1whatleytay for all help.
thx also to @VelocityRa for review and help me improve code.
For user want test this PR, i have seet 4 size icon diferent, Small/Medium/Big/Original
Vita3K create news line for first time launch inside config.yml file, just need change value: 0 to 4 for change icon size. 
Eg with Big value:
image

### CODE REVIEW

File: src/emulator/gui/src/game_selector.cpp [hunk 1]
```diff
@@ -44,25 +44,35 @@ void draw_game_selector(HostState &host) {
             ImVec2(0, 0), display_size);
     }
 
+    float icon_size = 64.0f;
```
Comment (pent0) [inline src/emulator/gui/src/game_selector.cpp]: Create a constant for this perhaps:

```cpp
const float default_icon_size = 64.0f

icon_size = default_icon_size;

case 1: icon_size = icon_size / 2;
case 2: icon_size = icon_size * 2;
```

File: src/emulator/gui/src/gui.cpp [hunk 2]
```diff
@@ -134,21 +150,33 @@ static void init_background(State &gui, const std::string &image_path) {
...
-        LOG_INFO("Could not load background from {}.", image_path);
+        LOG_INFO("Could not load image from {}.", image_path);
         return;
     }
 
-    GLuint texture;
-    glGenTextures(1, &texture);
-    gui.background_texture.init(texture, glDeleteTextures);
+    gui.background_texture.init(load_texture(width, height, data), glDeleteTextures);
+    delete data;
```
Comment (pent0) [inline src/emulator/gui/src/gui.cpp]:
```suggestion
    stbi_image_free(data);
```

File: src/emulator/gui/src/gui.cpp [hunk 3]
```diff
@@ -134,21 +150,33 @@ static void init_background(State &gui, const std::string &image_path) {
...
+            LOG_INFO("Could not load icon or image for title {}.", game.title_id);
+            continue;
+        }
+        stbi_uc *data = stbi_load_from_memory(&buffer[0], buffer.size(), &width, &height, nullptr, STBI_rgb_alpha);
+        if (width != 128 || height != 128) {
+            LOG_INFO("Invalid icon or image for title {}.", game.title_id);
+            continue;
+        }
+        host.gui.game_selector.icons[game.title_id].init(load_texture(width, height, data), glDeleteTextures);
+        delete data;
```
Comment (pent0) [inline src/emulator/gui/src/gui.cpp]:
```suggestion
        stbi_image_free(data);
```

File: src/emulator/gui/src/game_selector.cpp (hunk 1 above)
Comment (Zangetsu38) [reply src/emulator/gui/src/game_selector.cpp]: fixed :)

**Comment by pent0** [comment]
I'm not sure the GL parts are ok, so I will request another more review

File: src/emulator/config.cpp [hunk 4]
```diff
@@ -96,6 +96,7 @@ bool serialize(Config &cfg) {
     config_file_emit_single(emitter, "pstv-mode", cfg.pstv_mode);
     config_file_emit_single(emitter, "show-gui", cfg.show_gui);
     config_file_emit_single(emitter, "archive-log", cfg.archive_log);
+    config_file_emit_single(emitter, "size-icon", cfg.size_icon);
```
Comment (VelocityRa) [inline src/emulator/config.cpp]:
```suggestion
    config_file_emit_single(emitter, "icon-size", cfg.icon_size);
```

File: src/emulator/config.cpp [hunk 5]
```diff
@@ -131,6 +132,7 @@ static bool deserialize(Config &cfg) {
     get_yaml_value(config_node, "pstv-mode", &cfg.pstv_mode, false);
     get_yaml_value(config_node, "show-gui", &cfg.show_gui, false);
     get_yaml_value(config_node, "archive-log", &cfg.archive_log, false);
+    get_yaml_value(config_node, "size-icon", &cfg.size_icon, 2);
```
Comment (VelocityRa) [inline src/emulator/config.cpp]: ditto

File: src/emulator/gui/src/game_selector.cpp [hunk 6]
```diff
@@ -44,25 +44,38 @@ void draw_game_selector(HostState &host) {
             ImVec2(0, 0), display_size);
     }
 
+    constexpr float default_icon_size = 64.0f;
+    float icon_size = default_icon_size;
+
+    switch (host.cfg.size_icon){ 
+    case 1: icon_size = icon_size / 2; break;
```
Comment (VelocityRa) [inline src/emulator/gui/src/game_selector.cpp]: Don't use [magic values](https://www.wikiwand.com/en/Magic_number_(programming)), use an `enum` here, for example:
```c
enum IconSize {
    ICON_SIZE_SMALL,
    ICON_SIZE_DEFAULT,
    ICON_SIZE_MEDIUM,
    ICON_SIZE_LARGE,
};
```
put this enum somewhere that you can include both from here and from `config.cpp`
and cast the int to it here in the `switch` like so
```cpp
const auto icon_size = static_cast<IconSize>(host.cfg.icon_size);
switch (icon_size) {
```

File: src/emulator/config.cpp (hunk 4 above)
Comment (Zangetsu38) [reply src/emulator/config.cpp]: i have inversed for no have two time same value lol

Comment (VelocityRa) [reply src/emulator/config.cpp]: same value where?
in any case, change this and use a slightly different name in the other place (you can just add `_` at the end) but not here

Comment (Zangetsu38) [reply src/emulator/config.cpp]: i speak for float icon_size, i have set that for no confunded both ^^

**Comment by Zangetsu38** [comment]
no, but i have check all if used clang format ^^

**Comment by VelocityRa** [comment]
Not sure what you mean. Please run the script and ensure it doesn't change anything.

File: src/emulator/host/include/host/state.h [hunk 7]
```diff
@@ -89,3 +89,10 @@ struct HostState {
     GDBState gdb;
 #endif
 };
+
+enum IconSize {
```
Comment (VelocityRa) [inline src/emulator/host/include/host/state.h]: This is unrelated to the host, it doesn't belong here. Place it in `gui/state.h`

File: src/emulator/config.cpp [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 #include "config.h"
 
+#include <host/state.h>
```
Comment (VelocityRa) [inline src/emulator/config.cpp]:
```suggestion
#include <gui/state.h>
```

File: src/emulator/host/include/host/state.h (hunk 7 above)
Comment (Zangetsu38) [reply src/emulator/host/include/host/state.h]: oki

**Comment by Zangetsu38** [comment]
Done for all, i have also added news sizer, and fixed warning with using + 32, and no * 1.5, it no like virgule i think
